frontend/src/components/widgets/inputs/FontInput.vue

```diff
@@ -1,17 +1,26 @@
 <template>
 	<LayoutRow class="font-input">
-		<LayoutRow class="dropdown-box" :class="{ disabled }" :style="{ minWidth: `${minWidth}px` }" @click="() => !disabled && (open = true)" data-hover-menu-spawner>
-			<span>{{ activeEntry?.label || "" }}</span>
+		<LayoutRow class="dropdown-box" :class="{ disabled }" :style="{ minWidth: `${minWidth}px` }" tabindex="0" @click="toggleOpen" @keydown="keydown" data-hover-menu-spawner>
+			<span>{{ activeEntry?.name || "" }}</span>
 			<IconLabel class="dropdown-arrow" :icon="'DropdownArrow'" />
 		</LayoutRow>
-		<MenuList
-			v-model:activeEntry="activeEntry"
-			v-model:open="open"
-			@naturalWidth="(newNaturalWidth: number) => (minWidth = newNaturalWidth)"
-			:entries="entries"
-			:direction="'Bottom'"
-			:scrollableY="true"
-		/>
+		<FloatingMenu class="menu-list" v-model:open="open" :direction="'Bottom'" :type="'Dropdown'" ref="floatingMenu" :windowEdgeMargin="0" data-hover-menu-keep-open>
+			<LayoutCol :scrollableY="true" @scroll="onScroll" :style="{ width: `${minWidth}px` }" ref="scroller">
+				<LayoutRow class="spacer" :style="{ height: `${startIndex * 20}px` }"></LayoutRow>
+				<LayoutRow
+					v-for="(entry, entryIndex) in entries.slice(startIndex, endIndex)"
+					:key="entryIndex + startIndex"
+					class="row"
+					:class="{ active: entry === highlighted }"
+					@click="selectFont(entry.name)"
+				>
+					<link v-if="!isStyle" rel="stylesheet" :href="entry.url?.toString()" />
+
+					<span class="entry-label" :style="{ fontFamily: `${isStyle ? 'inherit' : entry.name}` }">{{ entry.name }}</span>
+				</LayoutRow>
+				<LayoutRow class="spacer" :style="{ height: `${totalHeight - endIndex * 20}px` }"></LayoutRow>
+			</LayoutCol>
+		</FloatingMenu>
```

Comment:
I'm concerned about getting rid of the usage of MenuList which aims to avoid code duplication by encapsulating the concepts of list rendering. Can you add the new features needed here directly to MenuList rather than replicating MenuList's behavior and modifying it here?

Reply:
Converting the only showing visible parts of the list requires that all MenuLists have a known size.

Reply:
Can that be only a requirement when enabled via a prop?

Reply:
What if we wanted a search box in the floating menu for just the font input - wouldn't it make sense to separate it out?

Reply:
I still think that would be better to include in the same MenuList code. It's fine if that's a very feature-rich component with lots of props to enable the many optional features. It's better than duplicating code. Otherwise we wouldn't have MenuList at all, we'd just use FloatingMenu for the menu bar, dropdowns, etc.

Reply:
This makes the MenuList extremely bloated, extremely convoluted and extremely difficulty to work with.

Reply:
I just find it difficult to work with single vue components that can have a lot of different behaviours.

Reply:
I think I managed to do this but I've probably broken some things.

Reply:
`MenuList.vue` seems pretty clean to me! The other excellent advantage is that you just implemented virtual scrolling, not only for fonts, but for all types of menu lists which is something we would have needed to do someday anyways.